src/libraries/System.Text.Json/Common/JsonSeparatorNamingPolicy.cs

```diff
@@ -21,149 +28,148 @@ public sealed override string ConvertName(string name)
                 ThrowHelper.ThrowArgumentNullException(nameof(name));
             }
 
-            // Rented buffer 20% longer that the input.
-            int rentedBufferLength = (12 * name.Length) / 10;
-            char[]? rentedBuffer = rentedBufferLength > JsonConstants.StackallocCharThreshold
-                ? ArrayPool<char>.Shared.Rent(rentedBufferLength)
-                : null;
-
-            int resultUsedLength = 0;
-            Span<char> result = rentedBuffer is null
-                ? stackalloc char[JsonConstants.StackallocCharThreshold]
-                : rentedBuffer;
+            return ConvertNameCore(_separator, _lowercase, name);
+        }
 
-            void ExpandBuffer(ref Span<char> result)
-            {
-                char[] newBuffer = ArrayPool<char>.Shared.Rent(result.Length * 2);
+        private static string ConvertNameCore(char separator, bool lowercase, string name)
+        {
+            Debug.Assert(name != null);
 
-                result.CopyTo(newBuffer);
+            char[]? rentedBuffer = null;
 
-                if (rentedBuffer is not null)
-                {
-                    result.Slice(0, resultUsedLength).Clear();
-                    ArrayPool<char>.Shared.Return(rentedBuffer);
-                }
+            // Rented buffer 20% longer that the input.
```

Comment:
```suggestion
            // Rented buffer 20% longer than the input.
```

Where did the 20% number come from? Is that arbitrary / a guess as to how much it might expand? It'd be worth expanding the comment a bit to explain its genesis.

Reply:
Long story short, I used the median length of words in English as a base. It was discussed somewhere in https://github.com/dotnet/corefx/pull/41354 I guess.